### PR TITLE
Multiple Client support

### PR DESCRIPTION
**Pull request by vlad-iancu**

### CODE REVIEW

File: vlad_task3/src/ServerProxyObject.cpp [hunk 1]
```diff
@@ -3,20 +3,37 @@
...
+        *available = value;
     }
 
     void ServerProxyObject::run()
     {
         uint8_t command;
         uint64_t recvd;
         uint32_t priority;
+        setServerAvailable(true);
         while (active)
```
Comment (maxlig79) [inline vlad_task3/src/ServerProxyObject.cpp:35]: Unfortunately the exit is not correct implemented: if one client send the exit command also the server will exit.

File: vlad_task3/include/Common.hpp [hunk 2]
```diff
@@ -14,7 +14,8 @@ namespace common
         DELETE = 2,
         GET = 3,
         EXIT = 4,
-        HELP = 5
+        HELP = 5,
+        ID = 6
```
Comment (maxlig79) [inline vlad_task3/include/Common.hpp:18]: You have added a command for this task, looks a little complicated.

File: vlad_task3/main_server.cpp [hunk 3]
```diff
@@ -11,11 +11,9 @@
 
 using namespace common;
 
-DynamicArray arr;
-
 int main()
 {
-    server::ServerProxyObject serv(&arr);
+    server::ServerProxyObject serv(5);
```
Comment (maxlig79) [inline vlad_task3/main_server.cpp]: What is 5 represents? Maybe you should use a macro or const.

File: vlad_task3/src/ClientProxyObject.cpp [hunk 4]
```diff
@@ -2,54 +2,56 @@
...
+#include <stdexcept>
 
 using namespace common;
 
 namespace client
 {
-    ClientProxyObject::ClientProxyObject(): ProxyBase(boost::interprocess::open_only)
+    ClientProxyObject::ClientProxyObject() : 
+    ProxyBase(boost::interprocess::open_only),
+    timeout(boost::posix_time::millisec(10))
```
Comment (maxlig79) [inline vlad_task3/src/ClientProxyObject.cpp]: Where is this timeout used?

File: vlad_task3/src/ClientProxyObject.cpp [hunk 5]
```diff
@@ -58,13 +60,50 @@ namespace client
...
+        int code = ID;
+        setId(id);
         mq->send(&code, sizeof(int), 0);
-        cond->wait(lk);
+        id_cond->wait(mem_lk);
+        if (!getServerAvailable())
+        {
+            throw std::runtime_error("Server is not available");
+        }
+        int clientId = getId();
```
Comment (maxlig79) [inline vlad_task3/src/ClientProxyObject.cpp]: Basically here the client is waiting for an id from the server which is pretty complicated.
Wouldn't be easier to add the client id to the send command(please check the other colleague work)?

**Review by maxlig79 — CHANGES_REQUESTED** [review]

**Review by maxlig79 — APPROVED** [review]

**Review by MariaCraciun1 — APPROVED** [review]

**Review by maxlig79 — APPROVED** [review]

**Review by maxlig79 — APPROVED** [review]

**Pull request merged**